valhalla/thor/unidirectional_astar.h

```diff
@@ -37,6 +35,36 @@ class UnidirectionalAStar : public PathAlgorithm {
    */
   explicit UnidirectionalAStar(const boost::property_tree::ptree& config = {});
 
+  std::vector<std::vector<PathInfo>>
+  GetBestPathDepartAt(valhalla::Location& origin,
+                      valhalla::Location& dest,
+                      baldr::GraphReader& graphreader,
+                      const sif::mode_costing_t& mode_costing,
+                      const sif::TravelMode mode,
+                      const Options& options = Options::default_instance()) override;
+
+  std::vector<std::vector<PathInfo>>
+  GetBestPathArriveBy(valhalla::Location& origin,
+                      valhalla::Location& dest,
+                      baldr::GraphReader& graphreader,
+                      const sif::mode_costing_t& mode_costing,
+                      const sif::TravelMode mode,
+                      const Options& options = Options::default_instance()) override;
+
+  /**
+   * Clear the temporary information generated during path construction.
+   */
+  void Clear() override;
```

Comment:
The diff is a bit weird here, but the essence is: 
 - created two new methods `GetBestPathDepartAt` and `GetBestPathArriveBy` 
 - moved the old `GetPath` to the `protected` section and templatized it
 - renamed the algorithm in `name()` to `time_dependent_a*` 
 - templatized some more methods 
 - split `FormPath` to `FormPathForward` and `FormPathReverse` because they share absolutely no code